CBinding/Gui/CTextEditorExtension.cs

```diff
@@ -535,11 +536,11 @@ string GetIndent (IReadonlyTextDocument d, int lineNumber, int terminateIndex)
 		[CommandHandler (MonoDevelop.DesignerSupport.Commands.SwitchBetweenRelatedFiles)]
 		protected void Run ()
 		{
-			var cp = (CProject)DocumentContext.Project;
+			var cp = (CMakeProject)(DocumentContext.Project as SolutionItem);
 			if (cp != null) {
 				string match = cp.MatchingFile (this.DocumentContext.Name);
 				if (match != null)
-					IdeApp.Workbench.OpenDocument (match, cp, true);
+					IdeApp.Workbench.OpenDocument (match, DocumentContext.Project, true);
```

Comment:
DocumentContext.Project here will be null.

Probably need an overload of OpenDocument that accepts a SolutionItem so you can have the document load with the CMakeProject as its owner.